test/functional/org_admin/plans_controller_test.rb

```diff
@@ -0,0 +1,84 @@
+require 'test_helper'
+
+class PlansControllerTest < ActionDispatch::IntegrationTest
+
+  include Devise::Test::IntegrationHelpers
+
+  setup do
+    # Get the first Org Admin
+    @org = Org.funder.first
+    @admin = User.create!(email: "org-admin-plans-tester@example.com", 
+                         firstname: "Org", surname: "Admin",
+                         password: "password123", password_confirmation: "password123",
+                         org: @org, accept_terms: true, confirmed_at: Time.zone.now)
+                         
+    # Make sure the user is an org admin
+    @admin.perms << Perm.where(name: ['grant_permissions', 'modify_guidance', 
+                                      'modify_templates', 'change_org_details'])
+    @admin.save!
+    
+    @regular_user = User.create!(email: 'org_admin_plans_tester@example.com', firstname: "Tester", surname: "Testing",
+                                 password: "password123", password_confirmation: "password123",
+                                 org: @org, accept_terms: true, confirmed_at: Time.zone.now, ) 
+    @plan = Plan.create!(template: Template.first, title: 'Test Plan', visibility: :privately_visible, feedback_requested: true,
+                         roles: [Role.new(user: @regular_user, creator: true)])
+    Role.create!(user: @admin, plan: @plan, access: Role.access_values_for(:reviewer).min)
+  end
+
+  test "unauthorized user cannot access the plans page" do
+    # Should redirect user to the root path if they are not logged in!
+    get org_admin_plans_path
+    assert_unauthorized_redirect_to_root_path
+    # Non Org-Admin cannot perform this action
+    sign_in @regular_user
+    get org_admin_plans_path
+    assert_authorized_redirect_to_plans_page
+  end
+  
+  test "org admin can access the plans page" do
+    sign_in @admin
+    get org_admin_plans_path
+
+    assert_response :success
+    assert assigns(:plans)
+    assert assigns(:feedback_plans)
```

Comment:
we should stop using assigns since enforces the test to know implementation details of that controller action moreover I think it is deprecated in Rails 5

Reply:
Yes agreed. @bhavi and have also noticed some issues with the way the tests use the seeds.rb file. It can create confusing and sometimes unpredicatable results when the tests run on Travis. They recently updated their env image, so I'm not sure if its do to that or not. We should move away from using the seed.rb and move towards Rails' fixtures or some mock data that is defined in the test_helper.rb
We need to have a team discussion about tests after MVP in March. I think it should be a priority.